scipy/stats/tests/test_stats.py

```diff
@@ -7361,6 +7415,72 @@ def test_inf_values(self):
             assert_equal(
                 stats.wasserstein_distance([1, 2, np.inf], [np.inf, 1]),
                 np.nan)
+        uv, vv, uw = [[1, 1], [2, 1]], [[np.inf, -np.inf]], [1, 1]
+        distance = stats.wasserstein_distance(uv, vv, uw)
+        assert_equal(distance, np.inf)
+        with suppress_warnings() as sup:
+            sup.record(RuntimeWarning, "invalid value*")
+            uv, vv = [[np.inf, np.inf]], [[np.inf, -np.inf]]
+            distance = stats.wasserstein_distance(uv, vv)
+            assert_equal(distance, np.nan)
+
+    @pytest.mark.parametrize('nu', (10, 15, 20))
+    @pytest.mark.parametrize('nv', (10, 15, 20))
+    @pytest.mark.parametrize('ndim', (1, 3, 5))
+    def test_multi_dim_nD(self, nu, nv, ndim):
+        # Adding dimension on distributions do not affect the result
+        rng = np.random.default_rng(2736495738494849509)
+        u_values = rng.random(size=(nu, ndim))
+        v_values = rng.random(size=(nv, ndim))
+        u_weights = rng.random(size=nu)
+        v_weights = rng.random(size=nv)
+        ref = stats.wasserstein_distance(u_values, v_values, u_weights, v_weights)
+
+        add_dim = rng.integers(0, ndim, size = 1)[0]
+        add_value = rng.random()
+
+        u_values = np.c_[u_values[:, :add_dim], 
+                         np.full(nu, add_value), 
+                         u_values[:, add_dim:]]
+        v_values = np.c_[v_values[:, :add_dim], 
+                         np.full(nv, add_value), 
+                         v_values[:, add_dim:]]
```

Comment:
```suggestion
        u_values = np.insert(u_values, add_dim, add_value, axis=1)
        v_values = np.insert(v_values, add_dim, add_value, axis=1)
```
Please use this idea in `test_zero_weight_nD`, too.